src/shared/hooks/useBreadcrumbs.tsx

```diff
@@ -100,11 +99,11 @@ export const useBreadcrumbs = (restCrumbs?: Breadcrumb[]) => {
     }
 
     setCrumbs([...newBreadcrumbs, ...(restCrumbs || [])]);
-  }, [t, firstName, lastName, appletsFoldersData, id, appletId, activityId]);
+  }, [t, firstName, lastName, appletsFoldersData, id, appletId, activityId, appletLabel, pathname]);
 
   useEffect(() => {
```

Comment:
Is it possible to remove the useeefect and usestate and make just dispatch on the top?